packages/app/pages/course/[cid]/insights.tsx

```diff
@@ -0,0 +1,207 @@
+
+  return (
+    <>
+      <StandardPageContainer>
+        <NavBar courseId={Number(cid)} />
+        <Row
+          align={"middle"}
+          justify={"space-between"}
+          style={{ margin: "12px 0px" }}
+        >
+          <h1 style={{ display: "inline", margin: "0px" }}>
+            Insights Dashboard
+          </h1>
+          <Row>
+            <div style={{ maxWidth: "200 px" }}>
+              <Tooltip
+                title={
+                  "If no date range is selected results are from the data for the full semester so far"
+                }
+              >
+                <QuestionCircleOutlined />
+              </Tooltip>
+              <b
+                style={{
+                  display: "inline-block",
+                  marginRight: "12px",
+                  marginLeft: "8px",
+                }}
+              >
+                Date Range
+              </b>
+              <RangePicker
+                onChange={(_, dateString) =>
+                  setDateRange({ start: dateString[0], end: dateString[1] })
+                }
+              />
+            </div>
+            <Button
+              style={{ marginLeft: "24px" }}
+              onClick={() => setSettingsVisible(true)}
+            >
+              Edit Insights
+            </Button>
+          </Row>
+        </Row>
+        <Divider style={{ margin: "0 0 16px 0" }} />
+        <Drawer
+          title="Display Options"
+          placement="left"
+          closable={true}
+          destroyOnClose={true}
+          onClose={() => setSettingsVisible(false)}
+          visible={settingsVisible}
+          width={400}
+        >
+          <InsightsDisplayOptions />
+        </Drawer>
+        <InsightsRowContainer>
+          {smallInsights?.map((insightName: string) => {
+            return (
+              <RenderInsight
+                key={insightName}
+                insightName={insightName}
+                dateRange={dateRange}
+              />
+            );
+          })}
+        </InsightsRowContainer>
+        <InsightsRowContainer>
+          {defaultInsights?.map((insightName: string) => {
+            return (
+              <RenderInsight
+                key={insightName}
+                insightName={insightName}
+                dateRange={dateRange}
+              />
+            );
+          })}
+        </InsightsRowContainer>
+      </StandardPageContainer>
+    </>
+  );
+}
+
+interface RenderInsightProps {
+  insightName: string;
+  dateRange: DateRangeType;
+}
+
+function RenderInsight({
+  insightName,
+  dateRange,
+}: RenderInsightProps): ReactElement {
+  const router = useRouter();
+  const { cid } = router.query;
+
+  const { data: insight } = useSWR(
+    cid &&
+      `api/v1/insights/${cid}/${insightName}?start=${dateRange.start}&end=${dateRange.end}`,
+    async () => await API.insights.get(Number(cid), insightName, dateRange)
+  );
+
+  if (!insight) {
+    // TODO: Create loading shell
```

Comment:
https://ant.design/components/spin/

Reply:
Turned out to be a bit of a refactor cause I wanted to have the loading spinner show inside the data insight card itself but it's all good now